src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -652,11 +659,15 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
     withWorkspaceContextAndPermissions(methodRepoQuery.source.workspaceName, WorkspaceAccessLevel.Read, txn) { workspaceContext =>
       withMethodConfig(workspaceContext, methodRepoQuery.source.namespace, methodRepoQuery.source.name, txn) { methodConfig =>
         import org.broadinstitute.dsde.rawls.model.MethodRepoJsonSupport._
-        RequestComplete(methodRepoDAO.postMethodConfig(
+        methodRepoDAO(context).postMethodConfig(
           methodRepoQuery.methodRepoNamespace,
           methodRepoQuery.methodRepoName,
           methodConfig.copy(namespace = methodRepoQuery.methodRepoNamespace, name = methodRepoQuery.methodRepoName),
-          userInfo))
+          userInfo) {
+
+          result => context.parent ! RequestComplete(result.get)
+        }
+        RequestComplete("done")
```

Comment:
Does this line serve a purpose other than satisfying the return type?  Would we still use `PerRequestMessage` if we moved over to this system?